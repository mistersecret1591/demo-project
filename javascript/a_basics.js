// Primitive arguments
function calculateCommissionJs(volume) {
    var fee, rate, commission, bonus, award;

    fee = 0.0;
    bonus = 0.0;
    if (volume > 200000.0) {
      rate = 0.15;
      bonus = 1000.0;
    } else {
      if (volume > 100000.0) rate = 0.123;
      else {
        if (volume > 50000.0) rate = 0.1;
        else {
          if (volume >= 10000.0) rate = 0.075;
          else {
            if (volume >= 5000.0) rate = 0.05;
            else {
              if (volume > 0.0) rate = 0.025;
              else {
                rate = 0;
                fee = 50.0;
              }
            }
          }
        }
      }
    }
    commission = Math.trunc(volume * rate * 100) / 100;
    award = commission + bonus - fee;
    return Math.trunc(award * 100) / 100;
}

// Structured arguments
function getClientAgeGenerationJs(client){
    const res = {
      gen: null,
      set: true
    }
    if (client.age <= 18){
        res.gen = "Gen.Z"
    } else if (client.age > 18 && client.age < 34){
        res.gen = "Gen.Y"
    } else if (client.age >= 34){
        res.gen = "Gen.X"
    } else {
        res.set = false
    }
    return res
}


// Structured arguments with deep nesting
function getClientScoreJs(client){
    if (client.age.firstname && client.age.lastname){
        var total_compliant_volume = 0;
        var dates = [];
        for (i = 0; i < client.orders.length; i++){
            if (client.orders[i].order_id != -1 && client.orders[i].currency === "EUR"){
                dates.push(client.orders[i].date)
                total_compliant_volume += client.orders[i].volume
            }
        }
        return total_compliant_volume / dates.length
    } else {
        return 0
    }
}

// Regex Inferer
function isEmailValidJs(email) {
    if (typeof email !== "string") {
      return false;
    }
    const unicodePattern = /[^\x00-\x7F]/;
    var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return !unicodePattern.test(email) && re.test(email.toLowerCase());
}