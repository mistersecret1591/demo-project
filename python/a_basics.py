# Primitive arguments
def calculate_commission_py(volume):
    fee = 0.0
    bonus = 0.0

    if (volume > 200000):
        rate = 0.15
        bonus = 1000
    else:
        if volume > 100000:
            rate = 0.123
        else:
            if volume > 50000:
                rate = 0.1
            else:
                if volume > 10000:
                    rate = 0.075
                else:
                    if volume > 5000:
                        rate = 0.05
                    else:
                        if volume > 0:
                            rate = 0.025
                        else:
                            rate = 0
                            fee = 50.0
    commission = int(volume * rate * 100) / 100.0
    award = commission + bonus - fee
    return award

# Structured arguments
def get_client_age_generation_py(client):
    res = {}
    res['gen'] = None
    res['set'] = True
    
    if client['age'] <= 18:
        res['gen'] = "Gen.Z"
    elif client.get("age") > 18 and client["age"] < 34:
        res['gen'] = "Gen.Y"
    elif client["age"] >= 34:
        res['gen'] = "Gen.X"
    else:
        res['set'] = False
    return res


# Structured arguments with deep nesting
def get_client_score_py(client):
    if (client.get('name')['firstname'] is not None) and (client['name']['lastname'] is not None):
        total_compliant_volume = 0
        dates = []
        for i in range(len(client["orders"])):
            if client["orders"][i]["order_id"] != -1 and client["orders"][i]["currency"] == "EUR":
                dates.append(client["orders"][i]["date"])
                total_compliant_volume += client["orders"][i]["volume"]
        return total_compliant_volume / len(dates)
    else:
        return 0


#Regex Inferer
import re
def isEmailValidPy(email):
    result = re.search(r"^[A-Za-z0-9\.\+_-]+@[A-Za-z0-9\._-]+\.[a-zA-Z]*$", email)
    return bool(result)
    
get_client_age_generation_py({'age': 12})