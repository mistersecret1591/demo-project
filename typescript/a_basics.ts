import { Client, ClientGen, GenResult } from "./a_basics.d";

// Primitive arguments
function calculateCommissionTs(volume: number) {
    var fee: number = 0.0;
    var rate: number;
    var commission: number;
    var bonus: number = 0.0;
    var award: number;

    if (volume > 200000.0) {
      rate = 0.15;
      bonus = 1000.0;
    } else {
      if (volume > 100000.0) rate = 0.123;
      else {
        if (volume > 50000.0) rate = 0.1;
        else {
          if (volume >= 10000.0) rate = 0.075;
          else {
            if (volume >= 5000.0) rate = 0.05;
            else {
              if (volume > 0.0) rate = 0.025;
              else {
                rate = 0;
                fee = 50.0;
              }
            }
          }
        }
      }
    }
    commission = Math.trunc(volume * rate * 100) / 100;
    award = commission + bonus - fee;
    return Math.trunc(award * 100) / 100;
}

// Structured arguments
function getClientAgeGenerationTs(client: ClientGen): GenResult {
    var res: GenResult = {
      gen: null,
      set: true
    };

    if (client.age <= 18){
        res.gen = "Gen.Z"
    } else if (client.age > 18 && client.age < 34){
        res.gen = "Gen.Y"
    } else if (client.age >= 34){
        res.gen = "Gen.X"
    } else {
        res.set = false
    }
    return res;
}


// Structured arguments with deep nesting
function getClientScoreTs(client: Client){
    if (client.name.firstname && client.name.lastname){
        var total_volume: number = 0;
        var dates: Array<String> = [];
        for (let i:number = 0; i < client.orders.length; i++){
            if (client.orders[i].order_id != "-1" && client.orders[i].currency === "EUR"){
                dates.push(client.orders[i].date)
                total_volume += client.orders[i].volume
            }
        }
        return total_volume / dates.length
    } else {
        return 0
    }
}

// Regex Inferer
function isEmailValidTs(email: string) {
    const unicodePattern = /[^\x00-\x7F]/;
    var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return !unicodePattern.test(email) && re.test(email.toLowerCase());
}