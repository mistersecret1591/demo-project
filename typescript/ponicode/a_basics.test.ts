import * as a_basics from "../a_basics"
// @ponicode
describe("a_basics.calculateCommissionTs", () => {
    test("0", () => {
        let result: any = a_basics.calculateCommissionTs(50000.0)
        expect(result).toMatchSnapshot()
    })

    test("1", () => {
        let result: any = a_basics.calculateCommissionTs(5000.0)
        expect(result).toMatchSnapshot()
    })

    test("2", () => {
        let result: any = a_basics.calculateCommissionTs(200001.0)
        expect(result).toMatchSnapshot()
    })

    test("3", () => {
        let result: any = a_basics.calculateCommissionTs(200000.0)
        expect(result).toMatchSnapshot()
    })

    test("4", () => {
        let result: any = a_basics.calculateCommissionTs(150000.0)
        expect(result).toMatchSnapshot()
    })

    test("5", () => {
        let result: any = a_basics.calculateCommissionTs(-Infinity)
        expect(result).toMatchSnapshot()
    })
})

// @ponicode
describe("a_basics.getClientAgeGenerationTs", () => {
    test("0", () => {
        let result: any = a_basics.getClientAgeGenerationTs({ age: 18 })
        expect(result).toMatchSnapshot()
    })

    test("1", () => {
        let result: any = a_basics.getClientAgeGenerationTs({ age: 35 })
        expect(result).toMatchSnapshot()
    })

    test("2", () => {
        let result: any = a_basics.getClientAgeGenerationTs({ age: 34 })
        expect(result).toMatchSnapshot()
    })

    test("3", () => {
        let result: any = a_basics.getClientAgeGenerationTs({ age: 18.0 })
        expect(result).toMatchSnapshot()
    })

    test("4", () => {
        let result: any = a_basics.getClientAgeGenerationTs({ age: 34.0 })
        expect(result).toMatchSnapshot()
    })

    test("5", () => {
        let result: any = a_basics.getClientAgeGenerationTs({ age: Infinity })
        expect(result).toMatchSnapshot()
    })
})

// @ponicode
describe("a_basics.isEmailValidTs", () => {
    test("0", () => {
        let result: any = a_basics.isEmailValidTs("user@host:300")
        expect(result).toMatchSnapshot()
    })

    test("1", () => {
        let result: any = a_basics.isEmailValidTs("bed-free@tutanota.de")
        expect(result).toMatchSnapshot()
    })

    test("2", () => {
        let result: any = a_basics.isEmailValidTs("something@example.com")
        expect(result).toMatchSnapshot()
    })

    test("3", () => {
        let result: any = a_basics.isEmailValidTs("user1+user2@mycompany.com")
        expect(result).toMatchSnapshot()
    })

    test("4", () => {
        let result: any = a_basics.isEmailValidTs("email@Google.com")
        expect(result).toMatchSnapshot()
    })

    test("5", () => {
        let result: any = a_basics.isEmailValidTs("")
        expect(result).toMatchSnapshot()
    })
})

// @ponicode
describe("a_basics.getClientScoreTs", () => {
    test("0", () => {
        let object: any = [{ order_id: "-1", currency: "EUR", date: "01-01-2030", volume: 0 }, { order_id: "-1", currency: "EUR", date: "01-13-2020", volume: -1 }, { order_id: "-1", currency: "EUR", date: "01-13-2020", volume: -1 }, { order_id: "-1", currency: "EUR", date: "01-13-2020", volume: -100 }]
        let result: any = a_basics.getClientScoreTs({ name: { firstname: "Michael", lastname: "Murray-Kynynmound" }, orders: object })
        expect(result).toMatchSnapshot()
    })

    test("1", () => {
        let object: any = [{ order_id: "-1", currency: "EUR", date: "01-13-2020", volume: 100 }]
        let result: any = a_basics.getClientScoreTs({ name: { firstname: "Michael", lastname: "Murray-Kynynmound" }, orders: object })
        expect(result).toMatchSnapshot()
    })

    test("2", () => {
        let object: any = [{ order_id: "-1", currency: "EUR", date: "01-01-2020", volume: 1 }, { order_id: "-1", currency: "EUR", date: "01-01-2020", volume: 1 }]
        let result: any = a_basics.getClientScoreTs({ name: { firstname: "Pierre Edouard", lastname: "Baziz" }, orders: object })
        expect(result).toMatchSnapshot()
    })

    test("3", () => {
        let object: any = [{ order_id: "-1", currency: "EUR", date: "01-13-2020", volume: 1 }]
        let result: any = a_basics.getClientScoreTs({ name: { firstname: "Jean-Philippe", lastname: "Dupont" }, orders: object })
        expect(result).toMatchSnapshot()
    })

    test("4", () => {
        let object: any = [{ order_id: "-1", currency: "EUR", date: "01-01-2030", volume: -100 }, { order_id: "-1", currency: "EUR", date: "01-13-2020", volume: -100 }, { order_id: "-1", currency: "EUR", date: "01-13-2020", volume: 0 }]
        let result: any = a_basics.getClientScoreTs({ name: { firstname: "George", lastname: "Baziz" }, orders: object })
        expect(result).toMatchSnapshot()
    })

    test("5", () => {
        let result: any = a_basics.getClientScoreTs({ name: { firstname: "", lastname: "" }, orders: [] })
        expect(result).toMatchSnapshot()
    })
})
