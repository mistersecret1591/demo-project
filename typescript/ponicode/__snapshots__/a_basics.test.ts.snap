// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a_basics.calculateCommissionTs 0 1`] = `3750`;

exports[`a_basics.calculateCommissionTs 1 1`] = `250`;

exports[`a_basics.calculateCommissionTs 2 1`] = `31000.15`;

exports[`a_basics.calculateCommissionTs 3 1`] = `24600`;

exports[`a_basics.calculateCommissionTs 4 1`] = `18450`;

exports[`a_basics.calculateCommissionTs 5 1`] = `NaN`;

exports[`a_basics.getClientAgeGenerationTs 0 1`] = `"Gen.Z"`;

exports[`a_basics.getClientAgeGenerationTs 1 1`] = `"Gen.X"`;

exports[`a_basics.getClientAgeGenerationTs 2 1`] = `"Gen.X"`;

exports[`a_basics.getClientAgeGenerationTs 3 1`] = `"Gen.Z"`;

exports[`a_basics.getClientAgeGenerationTs 4 1`] = `"Gen.X"`;

exports[`a_basics.getClientAgeGenerationTs 5 1`] = `"Gen.X"`;

exports[`a_basics.getClientScoreTs 0 1`] = `NaN`;

exports[`a_basics.getClientScoreTs 1 1`] = `NaN`;

exports[`a_basics.getClientScoreTs 2 1`] = `NaN`;

exports[`a_basics.getClientScoreTs 3 1`] = `NaN`;

exports[`a_basics.getClientScoreTs 4 1`] = `NaN`;

exports[`a_basics.getClientScoreTs 5 1`] = `0`;

exports[`a_basics.isEmailValidTs 0 1`] = `false`;

exports[`a_basics.isEmailValidTs 1 1`] = `true`;

exports[`a_basics.isEmailValidTs 2 1`] = `true`;

exports[`a_basics.isEmailValidTs 3 1`] = `true`;

exports[`a_basics.isEmailValidTs 4 1`] = `true`;

exports[`a_basics.isEmailValidTs 5 1`] = `false`;
